src/main/scala/sbtsonar/SonarPlugin.scala

```diff
@@ -41,19 +41,20 @@ object SonarPlugin extends AutoPlugin {
     sonarScan := {
       implicit val logger: Logger = streams.value.log
 
-      val maybeSonarHome = Option(System.getenv("SONAR_SCANNER_HOME"))
-        .orElse(Option(System.getProperty("sonarScanner.home")))
-
-      if (maybeSonarHome.isEmpty)
-        sys.error("SONAR_SCANNER_HOME environmental variable or sonarScanner.home system property not defined.")
+      val sonarHome = sys.env.get("SONAR_SCANNER_HOME")
+        .orElse(sys.props.get("sonarScanner.home"))
+        .getOrElse(sys.error("SONAR_SCANNER_HOME environment variable or sonarScanner.home system property not defined."))
 
       // Update the external properties file if the sonarUseExternalConfig is set to true.
       if (sonarUseExternalConfig.value)
         updatePropertiesFile(baseDirectory.value, SonarExternalConfigFileName, version.value)
 
-      val args = sonarScannerArgs(sonarUseExternalConfig.value, sonarProperties.value, version.value)
+      //Allow to set sonar properties via system properties: [https://docs.sonarqube.org/display/SONAR/Analysis+Parameters]
+      val mergedSonarProperties = sonarProperties.value ++ sys.env.filterKeys(_.startsWith("sonar."))
```

Comment:
Perhaps you were after `sys.props` instead of `sys.env`?

Reply:
Ouch, yes. It is my mistake. I talk about system properties. I will correct it .